installer/install.iss

```diff
@@ -74,7 +74,7 @@ PrivilegesRequired=none
 UninstallDisplayIcon={app}\{#MINGW_BITNESS}\share\git\git-for-windows.ico
 #ifndef COMPILE_FROM_IDE
 #if Pos('-',APP_VERSION)>0
-VersionInfoVersion={#Copy(APP_VERSION,1,Pos('-',APP_VERSION)-1)}
+VersionInfoVersion=GetFileVersion('{#MINGW_BITNESS}\bin\git.exe')
```

Comment:
Are you sure that this is executed? I could imagine that the `APP_VERSION` constant is defined somehow without a dash...

Reply:
Yes. But `APP_VERSION` is defined through `installer/release.sh`, right? This was just to make you remember what you said (which I think is not the correct approach). According to your word [that day](https://colabti.org/irclogger/irclogger_log/git-devel?date=2019-02-20#l61), you are actually changing the version of the installer, according to what (binary) is provided with git-sdk.

Reply:
Well, the wish was for the file version in the properties of the installer executable to match the file version in the properties of the included `git.exe`, and I kind of agree...

And yes, `APP_VERSION` is defined in `config.iss` that is written by `installer/release.sh`. But is it not also used for the installer's window title? IOW `APP_VERSION` is used, but we may not want to use it for the file version of the `.exe`.

If you change this so that `VersionInfoVersion` is defined independently from `APP_VERSION`, but using `GetFileVersion()` instead, does it work?

Reply:
> But is it not also used for the installer's window title?

Yes, ofc, it is used for window's title and other stuff (regedit). But won't it be a bad **UX** if user see different versions, I mean, on title bar and installer/installed bin? What was the intended use of `APP_VERSION` again? Can't we change the version of file while packing?

> If you change this so that `VersionInfoVersion` is defined independently from `APP_VERSION`, but using `GetFileVersion()` instead, does it work?

I can make `GetFileVersion()` work, but again, it will face the formentioned problem. Also how can we generate `git-sdk` using `git-sdk`? (basically mingw64\bin\git.exe)

Reply:
The file version is limited to four numbers separated by dots, IIUC, the version displayed in the window title is free form. So it can be a lot more expressive.

Reply:
@dscho my main concern here is **UX**, I mean if we are showing some different version on the title bar and installing some other version then this is bad(I know that we'll not be shipping such installer but yeah :P).

I'm making it to change the `VersionInfoVersion` to match the binary included and I guess we shall not be needing the conditional here because then we will get the same problematic installer when using `sdk build (installer|git-and-installer)` or `installer/release.sh <version-string>`?

https://github.com/git-for-windows/build-extra/blob/bbe200d9a0faace5249d912857a3968f90554b03/installer/install.iss#L76-L80

Reply:
Sounds good.